docs/proposals/user-impersonation-proposal.md

```diff
@@ -0,0 +1,39 @@
+# User Impersonation Proposal
+
+## Introduction
+We will need to add some security settings to help administrators lock down the broker. For this, we are going to check for privilege escalation and will move the location of where the APB pod will run.
+
+## Problem Description
+The problem is that currently privilege escalation is a concern for users who have access to the broker. We want to give the cluster admin who is setting up the service broker to have assurances that the broker can have some safety assurances.  
+
+## Implementation Details
+We will need to do 5 things to satisfy the requirements.
+1. Check if the user has the permissions to cover the cluster role to be used.
+2. If the cluster admin while setting up the broker has set a config value to auto escalate we will immediately continue and not check the user's permissions.
+3. Create transient namespace/project
+4. Create a service account with permissions to the target namespace/project.
+5. Create a pod running as the service account created above.
+6. Delete service account, role binding, and transient namespace/project.
+
+## Work Items
+- Create a new config value called auto escalate.
+- Use the [SubjectRulesReview](https://docs.openshift.org/latest/rest_api/apis-authorization.openshift.io/v1.SubjectRulesReview.html) API to retrieve the rules for a user.
```

Comment:
Looks like there's no kubernetes equivalent just yet: https://github.com/kubernetes/kubernetes/issues/31292 . Since we're adding Kubernetes support to the broker, we're going to need to find away to make this work or disable it until 1.8 has this feature.

Reply:
@jwmatthews have any thoughts. I think the answer is that if we are going to work on kubernetes that we you must opt into the auto-escalation?

Reply:
@shawn-hurley agree

Let's implement a sanity check during startup for configuration options.

If "auto_escalate: False" is set on a pure kubernetes run then the Broker should log the issue (unsupported configuration option on pure k8s)  error out and die.   I'm envisioning we'd perform this check during initial startup of broker and error out immediately if we are configured for an option we can not support. 

@shawn-hurley please  file a new issue on our broker to track the problem running on k8s lacking needed APIs for "auto_escalate: False",  include the link @rthallisey provided as a required dependency,  https://github.com/kubernetes/kubernetes/issues/31292